src/controllers/UserAuthenticationAjax.php

```diff
@@ -0,0 +1,256 @@
+<?php
+
+
+/**
+ * UserAuthenticationAjax
+ *
+ * @category    Erdiko
+ * @package     User
+ * @copyright   Copyright (c) 2016, Arroyo Labs, http://www.arroyolabs.com
+ * @author      Julian Diaz, julian@arroyolabs.com
+ */
+
+namespace erdiko\users\controllers;
+
+use erdiko\authenticate\BasicAuth;
+use erdiko\authenticate\iErdikoUser;
+
+use erdiko\users\models\User;
+use erdiko\users\models\Mailgun;
+
+class UserAuthenticationAjax extends \erdiko\core\AjaxController
+{
+
+    /**
+     * @param null $var
+     *
+     * @return mixed
+     */
+    public function get($var = null)
+    {
+        $this->id = 0;
+        if (!empty($var)) {
+            $routing = explode('/', $var);
+            if(is_array($routing)) {
+                $var = array_shift($routing);
+                $this->id = empty($routing)
+                    ? 0
+                    : array_shift($routing);
+            } else {
+                $var = $routing;
+            }
+
+            header('Content-Type: application/json');
+            return $this->_autoaction($var, 'get');
+        } else {
+            return $this->getNoop();
+        }
+    }
+
+    /**
+     * @param null $var
+     *
+     * @return mixed
+     */
+    public function post($var = null)
+    {
+        $this->id = 0;
+        if (!empty($var)) {
+            $routing = explode('/', $var);
+            if(is_array($routing)) {
+                $var = array_shift($routing);
+                $this->id = empty($routing)
+                    ? 0
+                    : array_shift($routing);
+            } else {
+                $var = $routing;
+            }
+
+            // load action based off of naming conventions
+            header('Content-Type: application/json');
+            return $this->_autoaction($var, 'post');
+        } else {
+            return $this->getNoop();
+        }
+    }
+
+    /**
+     * Default response for no action requests
+     */
+    protected function getNoop()
+    {
+        $response = array(
+            "action" => "None",
+            "success" => false,
+            "error_code" => 404,
+            "error_message" => 'Sorry, you need to specify a valid action'
+        );
+
+        $this->setContent($response);
+    }
+
+
+    public function postLogin()
+    {
+        $response = array(
+            "method" => "login",
+            "success" => false,
+            "error_code" => 0,
+            "error_message" => ""
+        );
+
+        try {
+            $data = json_decode(file_get_contents("php://input"));
```

Comment:
this does not always pick up POST/GET params. to see an example of this, use POSTman or just make a get request manually in the browser to see the lack of query string values.

please check all instances where we get params this way and make sure we also check the _REQUEST array (and strip tags from) to make sure we get data.